packages/amplify-schema-validator/src/index.ts

```diff
@@ -48,9 +54,9 @@ const allValidators = [
  * @param schemaString the graphql schema
  * @returns void
  */
-export const validateSchema = (schemaString: string): void => {
+export const validateSchema = (schemaString: string, amplifyFeatureFlags?: string, dataStoreEnabled?: boolean): void => {
```

Comment:
Rather than generating an ever-increasing list of options for validate, I'd prefer that we use an input type, which has all the fields. Also if we need something like dataStoreEnabled, we should require it everywhere (or explicitly only pass in the required fields per transformer, but requiring everywhere will likely be simpler given you have e2e control here).

So something like

```typescript
type SharedProps = {
  graphqlTransformerVersion: number;
  isDataStoreEnabled: boolean;
};

export type ValidateSchemaProps = SharedProps & {
  schemaString: string;
};

export type ParsedSchemaValidatorProps = SharedProps & {
  schema: DocumentNode;
};

export const validateSchema = (props: ValidateSchemaProps): void => {
  const schema = parse(props.schemaString);
  const validationErrors = allValidators.flatMap((validate) => validate({ ...props, schema }));
};
```

Reply:
Refactored